darts/utils/data/tabularization.py

```diff
@@ -161,3 +163,390 @@ def _create_lagged_data(
+        use as inputs.
+    target_lags
+        Optionally, the lags of the target series to be used as (auto-regressive) features. If not specified,
+        auto-regressive features will *not* be added to `X`.
+    lags_past_covariates
+        Optionally, the lags of `past_series` to be used as features.
+    lags_future_covariates
+        Optionally, the lags of `future_series` to be used as features.
+    lags_past_covariates
+        Optionally, the lags of `past_series` to be used as features.
+    lags_future_covariates
+        Optionally, the lags of `future_series` to be used as features.
+    max_samples
+        Optionally, the maximum number of samples to be drawn for training/validation; only the most recent
+        samples are kept. In theory, specifying a smaller `max_samples` should reduce computation time,
+        especially in cases where many observations could be generated.
+    multiple_outputs
+        Optionally, specifies whether the regression model predicts multiple timesteps into the future. If `True`,
+        then the regression model is assumed to predict all of the timesteps from time `t` to `t+horizon`. If
+        `False`, then the regression model is assumed to predict *only* the timestep at `t+horizon`.
+
+    Returns
+    -------
+    X
+        A `(n_observations, n_lagged_features, n_samples)`-shaped array of lagged features for training a
+        lagged-variables model.
+    y
+        If `multiple_outputs = True`, a `(n_observations, horizon, n_samples)`-shaped array of lagged
+        labels for training a lagged-variables model. Othewise, a `(n_observations, 1, n_samples)`-shaped
+        array of lagged labels.
+    times
+        The `time_index` of each observation in `X` and `y`.
+
+    Raises
+    ------
+    ValueError
+        If there are no times shared by all of the specified series, or if no lags have been specified.
+    TypeError
+        If the provided series do not share the same type of `time_index` (e.g. `target_series` uses a
+        pd.RangeIndex, but `future_series` uses a `pd.DatetimeIndex`).
+
+    References
+    ----------
+    .. [1] https://otexts.com/fpp2/AR.html#AR
+    .. [2] https://unit8.com/resources/time-series-forecasting-using-past-and-future-external-data-with-darts/
+    """
+
+    all_series = [target_series, past_series, future_series]
+    all_lags = [target_lags, past_lags, future_lags]
+    _check_lags(all_lags)
+    times, times_idxs = _find_intersecting_times(
+        all_series, all_lags, max_samples, horizon=horizon, is_training=True
+    )
+    X = []
+    for i, time_idx in enumerate(times_idxs):
+        if time_idx is not None:
+            # Offset shared time indices by requested lags:
+            idx_to_get = time_idx.reshape(-1, 1) + all_lags[i]
+            X.append(_create_lagged_vals(all_series[i], idx_to_get))
+    X = np.concatenate(X, axis=1)
```

Comment:
I think this part above to create X can benefit from the `create_lagged_features()` with an extra parameter indicating whether those features are being used for training or prediction (i.e., for the +1)